src/map/latent_effect_container.cpp

```diff
@@ -1785,3 +1769,33 @@ void CLatentEffectContainer::CheckLatentsZone(uint16 weather)
     }
     m_POwner->UpdateHealth();
 }
+
+/************************************************************************
+*																		*
+*  Checks all latents regarding current weather							*
+*																		*
+************************************************************************/
+
+void CLatentEffectContainer::CheckLatentsWeather(uint16 weather)
+{
+    for (uint16 i = 0; i < m_LatentEffectList.size(); ++i)
+    {
+        if (m_LatentEffectList.at(i)->GetConditionsID() == LATENT_WEATHER_ELEMENT)
+        {
+            // get the scholar weather and convert to a weather element
+            auto sch_weather = zoneutils::GetWeatherElement(battleutils::GetScholarWeather(m_POwner));
+
+            // if either the new weather or the scholar weather satisfy the latent then activate it
```

Comment:
hmm.. this is still wrong, though.  battleutils::GetWeather does it right, so I'd rather have that do it - might have to just add another parameter to it for zone weather override (or separate them and have one call the other using zone->getweather)